src/Bonsai.Scripting.Python/SetItem.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Reactive.Linq;
+using Pythonnet = Python.Runtime;
+using System.Xml.Serialization;
+using System.Linq;
+
+namespace Bonsai.Scripting.Python
+{
+    /// <summary>
+    /// Represents an operator that sets the named attribute of an object to a value.
+    /// </summary>
+    [WorkflowElementCategory(ElementCategory.Sink)]
+    [Combinator]
+    public class SetItem
+    {
+        [Description("The index to set.")]
+        public int Index { get; set; } = 0;
```

Comment:
Same comment re. variable types of indexers. The current implementation seems to only support `int`, but we know `string` also works, and possibly many other types. Flipping the logic around and making `PyObject` the property would keep things more flexible.